web/src/components/tasks/CheckElevatorRegistrationStatus.tsx

```diff
@@ -38,8 +40,38 @@ export const CheckElevatorRegistrationStatus = (props: Props): ReactElement => {
     } as Municipality;
   });
 
+  useEffect(() => {
+    if (business?.profileData.communityAffairsAddress?.streetAddress1 && formValues.address1 === "") {
+      formValues.address1 = business.profileData.communityAffairsAddress.streetAddress1;
+    }
+    if (business?.profileData.communityAffairsAddress?.streetAddress2 && formValues.address2 === "") {
+      formValues.address2 = business.profileData.communityAffairsAddress.streetAddress2;
+    }
+    if (business?.profileData.communityAffairsAddress?.municipality && !selectedMunicipality) {
+      setSelectedMunicipality(business.profileData.communityAffairsAddress.municipality);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
   const onSubmit = (event: FormEvent<HTMLFormElement>): void => {
     event.preventDefault();
+
+    if (business?.profileData && formValues.address1 && selectedMunicipality) {
+      const profileData = {
+        ...business?.profileData,
+        communityAffairsAddress: {
+          streetAddress1: formValues.address1,
+          streetAddress2: formValues.address2,
+          municipality: {
+            id: selectedMunicipality?.id,
+            name: selectedMunicipality?.name,
+            displayName: selectedMunicipality?.name,
+            county: selectedMunicipality?.county,
+          },
+        },
+      };
+      updateQueue?.queueProfileData(profileData).update();
```

Comment:
So based on the way this is written I assume the intention is to only save if you have both address1 and selectedMunicipality selected as opposed to any one field filled out and then saved in isolation. 

I just wanted to double check that this is desired functionality.